transport/tchannel/transport.go

```diff
@@ -212,3 +217,16 @@ func (t *Transport) stop() error {
 func (t *Transport) IsRunning() bool {
 	return t.once.IsRunning()
 }
+
+// onPeerStatusChanged receives notifications from TChannel Channel when any
+// peer's status changes.
+func (t *Transport) onPeerStatusChanged(tp *tchannel.Peer) {
+	t.lock.Lock()
+	defer t.lock.Unlock()
+
+	p, ok := t.peers[tp.HostPort()]
+	if !ok {
+		return
```

Comment:
It’s possible to receive a notification after a peer has been released, but in that case the notification is safe to drop.